### PR TITLE
Add some comments in engine headers/BXT files and hooked 'engineapi' for oldest WON builds

### PR DESCRIPTION
**Pull request by SmileyAG**
- Added comments in some of engine headers to show in which builds the struct ends on a specific function
- Hooked `engine_api_t` interface that removed starting from 3248 build or earlier, but at least exist in 1712 build (could be used  later for replacing pattern hooks of some functions with the calling functions from the structs itself)

### CODE REVIEW

File: HLSDK/engine/eiface.h [hunk 1]
```diff
@@ -252,7 +252,7 @@ typedef struct enginefuncs_s
 
 	void		(*pfnGetPlayerStats)		( const edict_t *pClient, int *ping, int *packet_loss );
 
-	void		(*pfnAddServerCommand)		( const char *cmd_name, void (*function) (void) );
+	void		(*pfnAddServerCommand)		( const char *cmd_name, void (*function) (void) ); // Smiley: Last function in 1712 build
```
Comment (YaLTeR) [inline HLSDK/engine/eiface.h:255]: Could you change these comments to be below (so like in-between)? Something like:

```
void (*old_function)();

// Functions below are present only in builds > 1712

void (*new_function)();
```

File: HLSDK/common/engine_launcher_api.h [hunk 2]
```diff
@@ -0,0 +1,101 @@
+// engine/launcher interface
```
Comment (YaLTeR) [inline HLSDK/common/engine_launcher_api.h:1]: Where does this file come from?

Comment (SmileyAG) [reply HLSDK/common/engine_launcher_api.h:1]: That header is exist in HLSDK 2.0-HLSDK 2.3, but got removed with release SDK to GitHub

**Comment by YaLTeR** [comment]
Thanks

**Pull request merged**